### PR TITLE
lz4: Fix pkgsStatic build

### PR DESCRIPTION
**Pull request by vaibhavsagar**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by Mic92** [comment]
Marked as WIP because of the todos.

File: pkgs/tools/compression/lz4/default.nix [hunk 1]
```diff
@@ -17,12 +19,17 @@ stdenv.mkDerivation rec {
 
   enableParallelBuilding = true;
 
-  makeFlags = [ "PREFIX=$(out)" "INCLUDEDIR=$(dev)/include" ];
+  makeFlags = [ "PREFIX=$(out)" "INCLUDEDIR=$(dev)/include" ]
+    # TODO always define both
+    ++ stdenv.lib.optional (!enableStatic) "BUILD_STATIC=no"
+    ++ stdenv.lib.optional (!enableShared) "BUILD_SHARED=no"
```
Comment (Mic92) [inline pkgs/tools/compression/lz4/default.nix]:
```
makeFlags = [ 
  "PREFIX=$(out)" 
  "INCLUDEDIR=$(dev)/include"
  "BUILD_STATIC=${if enableStatic then "yes" else "now"}"
  "BUILD_SHARED=${if enableShared then "yes" else "now"}"
]
```

**Comment by vaibhavsagar** [comment]
@Mic92 the TODOs are to prevent a mass rebuild, this is good to go as is.

**Comment by Mic92** [comment]
However it does prevent a rebuild.

**Comment by vaibhavsagar** [comment]
Advised to mark this WIP by @Ericson2314, changing back.

**Comment by vaibhavsagar** [comment]
@Mic92 better now?

**Comment by Mic92** [comment]
It is still a mass-rebuild because `makeFlags` gets extended. However you can also just remove the TODOs and move the commit to staging.

**Comment by Ericson2314** [comment]
@Mic92 my intention is to rebase this on top of the second-to-top branch commit of https://github.com/NixOS/nixpkgs/pull/51957 and merge into both stable and master for immediate usage, so I definitely don't want this to be a mass rebuild. We'll get it not to be shortly.

**Comment by Ericson2314** [comment]
OK both this and it's 18.09 equivalent, #51969, are just waiting on @ofborg. After they're merged, we can do the staging cleanup.

**Comment by vaibhavsagar** [comment]
Thanks!

**Pull request merged**